jobclient/python/cook/scheduler/client/jobs.py

```diff
@@ -112,197 +104,181 @@ class Job:
     command: str
     mem: float
     cpus: float
-
     uuid: UUID
     name: str
     max_retries: int
-    max_runtime: int
+    max_runtime: timedelta
     status: Status
     state: State
     priority: int
-    disable_mea_culpa_retries: bool
-
+    framework_id: str
+    retries_remaining: int
+    submit_time: datetime
+    user: str
     executor: Optional[Executor]
-    expected_runtime: Optional[int]
+    container: Optional[dict]
+    disable_mea_culpa_retries: Optional[bool]
+    expected_runtime: Optional[timedelta]
     pool: Optional[str]
     instances: Optional[List[Instance]]
     env: Optional[Dict[str, str]]
     uris: Optional[List[FetchableUri]]
-    container: Optional[dict]
     labels: Optional[Dict[str, str]]
     constraints: Optional[Set[Constraint]]
+    group: Optional[UUID]
     groups: Optional[List[UUID]]
     application: Optional[Application]
     progress_output_file: Optional[str]
     progress_regex_string: Optional[str]
-    user: Optional[str]
-    datasets: Optional[list]
-    gpus: Optional[float]
-    framework_id: Optional[str]
+    datasets: Optional[List[Dataset]]
```

Comment:
Support for data locality never finished getting implemented. I would say we leave this out for now.